Project not buildable on JDK 11 out of the box

Just checking out the project and running `mvn clean install` with JDK 11 gives:
```
[WARNING] 
java.lang.NoClassDefFoundError: javax/annotation/Generated
    at org.jboss.test.audit.generate.SectionsClassGenerator.generateSource (SectionsClassGenerator.java:121)
    at org.jboss.test.audit.generate.SectionsClassGenerator.generateToFile (SectionsClassGenerator.java:196)
    at org.jboss.test.audit.generate.SectionsClassGenerator.main (SectionsClassGenerator.java:83)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    at java.lang.Thread.run (Thread.java:834)
Caused by: java.lang.ClassNotFoundException: javax.annotation.Generated
    at java.net.URLClassLoader.findClass (URLClassLoader.java:471)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:589)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:522)
    at org.jboss.test.audit.generate.SectionsClassGenerator.generateSource (SectionsClassGenerator.java:121)
    at org.jboss.test.audit.generate.SectionsClassGenerator.generateToFile (SectionsClassGenerator.java:196)
    at org.jboss.test.audit.generate.SectionsClassGenerator.main (SectionsClassGenerator.java:83)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    at java.lang.Thread.run (Thread.java:834)
```

This is not very friendly and we should find the cause and fix it.

--- a/pom.xml
+++ b/pom.xml
@@ -113,8 +113,8 @@
         <jsf.api.version>3.0.0-RC1</jsf.api.version>
         <jaxrs.api.version>3.0.0</jaxrs.api.version>
         <!-- Test tools/dependencies -->
-        <jboss.test.audit.version>1.1.4.Final</jboss.test.audit.version>
         <testng.version>7.3.0</testng.version>
+        <jboss.test.audit.version>2.0.0.Final</jboss.test.audit.version>
         <arquillian.version>1.7.0.Alpha2</arquillian.version>
         <arquillian.container.se.api.version>1.0.1.Final</arquillian.container.se.api.version>
         <apache.httpclient.version>3.1</apache.httpclient.version>

